### PR TITLE
Use go1.16 for go interop tests and for newly added interop_matrix tests.

### PR DESCRIPTION
**Pull request by jtattermusch**
Fixes https://github.com/grpc/grpc/issues/26063

### CODE REVIEW

File: templates/tools/dockerfile/interoptest/grpc_interop_go1.16/Dockerfile.template [hunk 1]
```diff
@@ -0,0 +1,23 @@
+%YAML 1.2
+--- |
+  # Copyright 2015 gRPC authors.
```
Comment (dfawley) [inline templates/tools/dockerfile/interoptest/grpc_interop_go1.16/Dockerfile.template:3]: 2021?

File: tools/dockerfile/interoptest/grpc_interop_go1.16/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Copyright 2015 gRPC authors.
```
Comment (dfawley) [inline tools/dockerfile/interoptest/grpc_interop_go1.16/Dockerfile:1]: Same

File: tools/dockerfile/interoptest/grpc_interop_go1.16/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+RUN apt-get update && apt-get install -y \
+    python-all-dev \
+    python3-all-dev \
+    python-setuptools
+
+# Install Python packages from PyPI
+RUN curl https://bootstrap.pypa.io/pip/2.7/get-pip.py | python2.7
+RUN pip install --upgrade pip==19.3.1
+RUN pip install virtualenv==16.7.9
+RUN pip install futures==2.2.0 enum34==1.0.4 protobuf==3.5.2.post1 six==1.15.0 twisted==17.5.0
```
Comment (dfawley) [inline tools/dockerfile/interoptest/grpc_interop_go1.16/Dockerfile:34]: For my future reference: where did all these version numbers come from?  Are they all the same as the old version for Go or did you update them?

File: tools/dockerfile/interoptest/grpc_interop_go1.16/build_interop.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# Copyright 2015 gRPC authors.
```
Comment (dfawley) [inline tools/dockerfile/interoptest/grpc_interop_go1.16/build_interop.sh:2]: Year

**Review by dfawley — APPROVED** [review]
Thank you!

File: tools/dockerfile/interoptest/grpc_interop_go1.16/Dockerfile (hunk 3 above)
Comment (jtattermusch) [reply tools/dockerfile/interoptest/grpc_interop_go1.16/Dockerfile:34]: This file is generated from a template. So these numbers come from the `python_deps.include` file

https://github.com/grpc/grpc/blob/fb6669b17150e56f20f95f71b31f7055960affd3/templates/tools/dockerfile/interoptest/grpc_interop_go/Dockerfile.template#L20

**Pull request merged**